### PR TITLE
docs: add ADR + release process

### PR DESCRIPTION
**Pull request by ph--**
## Description

New documentation for ADR and Release process

## Type of Change

Please delete options that are not relevant:

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation update
- [ ] 🔧 Refactoring (no functional changes, no api changes)
- [ ] ⚡ Performance improvements
- [ ] 🧪 Test updates
- [ ] 🔒 Security improvements

## Changes Made

- New documentation to clarify the release process
The goal is to avoid making the same mistake I did by creating a latest release directly, and not a pre-release first (there was not assets linked to the release, the extension couldn't be installed anymore)

- Add an ADR section
The goal is to create ADR every time there's a breaking change, or a new feature.

## Testing

Please describe the tests that you ran to verify your changes:

- [ ] Unit tests pass: `go test ./...`
- [ ] Build succeeds: `go build -o gh-app-auth .`
- [ ] Manual testing performed
- [ ] Integration tests with real GitHub Apps (if applicable)

### Test Configuration

- **OS**:
- **Go version**:
- **GitHub CLI version**:

## Security Considerations

If this PR involves security-related changes:

- [ ] No sensitive data (tokens, keys) exposed in code or tests
- [ ] Input validation added/reviewed for new functionality  
- [ ] Private key handling follows security best practices
- [ ] Token caching security maintained
- [ ] No new attack vectors introduced

## Documentation

- [ ] Code comments updated
- [ ] README.md updated (if needed)
- [ ] Architecture documentation updated (if needed)
- [ ] Command help text updated (if needed)

## Checklist

- [ ] My code follows the project's style guidelines
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published
- [ ] **Commit messages follow [Conventional Commits](https://github.com/AmadeusITGroup/gh-app-auth/blob/main/CONTRIBUTING.md#commit-message-guidelines) format**
- [ ] **PR title follows conventional commits format** (e.g., `feat(auth): add token caching`)

## Breaking Changes

If this is a breaking change, please describe the impact and migration path for existing users:

## Additional Notes

Any additional information, concerns, or context for reviewers.

### CODE REVIEW

File: AGENTS.md [hunk 1]
```diff
@@ -332,7 +399,24 @@ test(auth): add integration tests for token refresh
...
 4. Update README.md command reference
-5. Run `make test` and `make quality`
+5. Ask the user whether the command warrants an ADR (it does if it introduces a new concept)
+6. Run `make test` and `make quality`
+
+### Adding a Feature or Making a Breaking Change
+
+1. Implement the change following the patterns in the files you touch
+2. Add or update tests covering the new behaviour and its edge cases
+3. Update `README.md`, `docs/`, and `CHANGELOG.md` (`[Unreleased]` section)
```
Comment (wherka-ama) [inline AGENTS.md:409]: A bit of a note to myself.
I'm less and less convinced about having an explicit CHANGELOG.md in git history.
It makes sense to have a relatively high level version in release notes, but when we start maintaining the log in git we're hitting a chicken and egg problem. We start treating the changes in CHANGELOG.md as documentation drifts. 
In short: it's a liability. 
I'd rather keep it somewhat externally and ideally it should be generated from a factual state of git history, not hand-written. I think we can afford it in the era of the agents.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -313,11 +313,24 @@ All submissions require code review. Please:
 
 ## Release Process
 
-Releases are automated through GitHub Actions:
+Releases are automated through GitHub Actions, and the trigger is a **GitHub pre-release** created on
```
Comment (wherka-ama) [inline CONTRIBUTING.md:316]: Another generic comment.

I started some time ago investigating the possibility to introduce immutable releases and adding more measures to establish a higher level of compliance with SLSA.
I think it is very important to finalise it. We are navigating the landscape which is closely linked with sensitive flows where the trust in a software delivery chain is of paramount importance.

File: docs/adr/README.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                 └──────► Superseded by NNNN
+```
+
+- **Proposed** — under discussion, typically in an open PR
+- **Accepted** — the decision is in effect and reflected in the code
+- **Deprecated** — no longer applies, with no direct replacement
+- **Superseded by NNNN** — replaced by a later ADR; keep the original text intact and add the pointer
+
+Never rewrite the body of an `Accepted` ADR to reflect a new decision. Change its status and write
+the new one.
```
Comment (wherka-ama) [inline docs/adr/README.md:76]: + Only leave a note of the ADR being superseded.
(only if it brings more clarity - leaving it with you to re-assess the need and proceed accordingly)

**Review by wherka-ama — APPROVED** [review]
@ph-- : thank you so much ❤️ That's super helpful indeed!
